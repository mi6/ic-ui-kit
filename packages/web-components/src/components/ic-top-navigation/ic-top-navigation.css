@import "../../global/normalize.css";

:host {
  display: block;
  color: var(--ic-theme-text);
  width: 100%;
  position: relative;
}

:host .top-navigation {
  background-color: var(--ic-theme-primary);
  height: max-content;
}

:host(.fullwidth-searchbar) slot[name="search"]::slotted(form) {
  width: 100%;
}

:host .title-link {
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  padding: var(--ic-space-xxs);
  transition: var(--ic-easing-transition-fast);
}

:host .title-link,
:host .title-link:visited,
:host .title-link:active,
:host .title-link ::slotted(a),
:host .title-link:visited ::slotted(a),
:host .title-link:active ::slotted(a) {
  color: var(--ic-theme-text);
  text-decoration: none;
  outline: none;
}

:host .title-link:hover:not(:focus) {
  background-color: var(--ic-theme-hover);
}

:host .title-link:active:not(:focus) {
  background-color: var(--ic-theme-active);
}

:host .title-link:hover {
  border-radius: var(--ic-border-radius);
}

:host .title-link:focus,
:host .title-link:focus-within {
  border-radius: var(--ic-border-radius);
  box-shadow: var(--ic-border-focus);
  outline: var(--ic-hc-focus-outline);
}

:host .title-link ic-typography {
  font-weight: 600;
}

:host .nav-panel-container {
  border-top: var(--ic-keyline-lighten);
  padding: 0 var(--ic-space-lg);
  margin-left: calc(-1 * var(--section-container-margin));
  margin-right: calc(-1 * var(--section-container-margin));
  display: flex;
}

:host(.dark) .nav-panel-container {
  border-top: var(--ic-keyline-darken);
}

.top-panel-container {
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.navigation-tabs {
  margin-top: calc(-1 * var(--ic-space-1px));
}

.app-details-container {
  display: flex;
  align-items: center;
  flex: 1 1 auto;
  margin-right: var(--ic-space-md);
}

.app-icon-container {
  display: flex;
  padding-right: var(--ic-space-xs);
}

.icon-buttons-container {
  display: flex;
  margin-left: var(--ic-space-md);
}

.icon-buttons-container ::slotted(nav) {
  display: flex;
}

.title-wrap {
  overflow-wrap: break-word;
  word-wrap: break-word;
  -ms-word-break: break-all;
  word-break: break-word;
  hyphens: auto;
}

:host .app-status {
  border-radius: var(--ic-space-md);
  background-color: var(--ic-architectural-white);
  color: var(--ic-color-primary-text);
  padding: var(--ic-space-xxs) var(--ic-space-lg);
  margin-left: var(--ic-space-md);
  flex: 1 1 0;
  max-width: fit-content;
}

:host(.dark) .app-status {
  background-color: var(--ic-theme-text);
  color: var(--ic-color-white-text);
}

:host .app-version {
  border-radius: var(--ic-space-md);
  background-color: var(--ic-theme-active);
  padding: var(--ic-space-xxs) var(--ic-space-sm);
  margin-left: var(--ic-space-xs);
  flex: 1 1 0;
  max-width: fit-content;
}

slot[name="app-icon"]::slotted(svg) {
  fill: var(--ic-theme-text);
  width: 2em;
  height: 2em;
}

slot[name="toggle-icon"] svg {
  fill: var(--ic-theme-text);
}

.search-menu-container {
  justify-content: right;
  display: flex;
  align-items: center;
}

.menu-button-container {
  margin-left: var(--ic-space-md);
}

.menu-button-container .navigation-landmark-button-text {
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.search-actions-container {
  display: flex;
}

.menu-buttons-slot {
  display: flex;
  flex-direction: column;
}

.navigation-landmark-text {
  position: absolute;
  width: var(--ic-space-1px);
  height: var(--ic-space-1px);
  padding: 0;
  margin: calc(-1 * var(--ic-space-1px));
  overflow: hidden;
}

.navigation-item-list {
  display: flex;
  list-style: none;
  height: 2.75rem;
}

.navigation-item-list::-webkit-scrollbar {
  display: none;
}

:host ic-section-container {
  padding-bottom: 0 !important;
  padding-top: 0 !important;
}

@media screen and (min-width: 993px) {
  .app-details-container {
    margin-right: var(--ic-space-lg);
  }
}

/* large */
@media screen and (max-width: 1200px) {
  :host .nav-panel-container {
    padding: 0 var(--ic-space-md);
  }
}

/* medium */
@media screen and (max-width: 992px) {
  :host .nav-panel-container,
  :host .app-status,
  :host .app-version {
    display: none;
  }

  :host .title-link {
    margin-right: var(--ic-space-xs);
  }

  .search-menu-container {
    max-width: 10rem;
  }

  .search-bar-container {
    display: flex;
    justify-content: center;
    align-items: center;
    border-top: var(--ic-keyline-darken);
    height: 4rem;
    padding-left: var(--ic-space-md);
    padding-right: var(--ic-space-md);
    margin-left: calc(-1 * var(--section-container-margin));
    margin-right: calc(-1 * var(--section-container-margin));
  }
}

/* x small */
@media screen and (max-width: 576px) {
  :host .title-link {
    margin-right: var(--ic-space-xxxs);
    word-break: break-word;
    hyphens: none;
  }

  .top-panel-container {
    min-height: 2.5rem;
  }

  .searchbox-inline {
    display: none;
  }

  .search-bar-container {
    margin-top: 0;
    height: 3.5rem;
    padding-left: var(--ic-space-xs);
    padding-right: var(--ic-space-xs);
  }

  .menu-button-container {
    margin-left: var(--ic-space-sm);
  }

  slot[name="app-icon"]::slotted(svg) {
    width: 1.5em;
    height: 1.5em;
  }

  slot[name="toggle-icon"] svg {
    width: 1.5em;
    height: 1.5em;
  }
}

@media (forced-colors: active) {
  :host .top-navigation {
    border-bottom: var(--ic-hc-border);
  }

  .app-status,
  .app-version {
    border: var(--ic-hc-border);
  }

  slot[name="app-icon"]::slotted(svg) {
    fill: currentcolor;
  }
}
