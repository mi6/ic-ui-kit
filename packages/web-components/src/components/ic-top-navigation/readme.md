# ic-top-navigation



<!-- Auto Generated Below -->


## Properties

| Property         | Attribute         | Description                                                                                     | Type                                 | Default        |
| ---------------- | ----------------- | ----------------------------------------------------------------------------------------------- | ------------------------------------ | -------------- |
| `appTitle`       | `app-title`       | The app title to be displayed. This is required, unless a slotted app title link is used.       | `string`                             | `undefined`    |
| `contentAligned` | `content-aligned` | The alignment of the top navigation content.                                                    | `"center" \| "full-width" \| "left"` | `"full-width"` |
| `href`           | `href`            | The URL to navigate to when the app title is clicked.                                           | `string`                             | `"/"`          |
| `inline`         | `inline`          | If `true`, the flyout navigation menu on small devices will be contained by the parent element. | `boolean`                            | `false`        |
| `shortAppTitle`  | `short-app-title` | The short title of the app to be displayed at small screen sizes in place of the app title.     | `string`                             | `""`           |
| `status`         | `status`          | The status info to be displayed.                                                                | `string`                             | `""`           |
| `version`        | `version`         | The version info to be displayed.                                                               | `string`                             | `""`           |


## Slots

| Slot            | Description                                                                         |
| --------------- | ----------------------------------------------------------------------------------- |
| `"app-icon"`    | Content will be rendered to left of app title.                                      |
| `"app-title"`   | Handle routing by nesting a route in the app title.                                 |
| `"buttons"`     | Content will be rendered to right of search bar.                                    |
| `"navigation"`  | Content will be rendered in navigation panel.                                       |
| `"search"`      | Content will be rendered in search area to left of buttons.                         |
| `"toggle-icon"` | Icon to be displayed on the button to toggle search slot content on smaller devices |


## Dependencies

### Depends on

- [ic-section-container](../ic-section-container)
- [ic-typography](../ic-typography)
- [ic-button](../ic-button)
- [ic-horizontal-scroll](../ic-horizontal-scroll)
- [ic-navigation-menu](../ic-navigation-menu)

### Graph
```mermaid
graph TD;
  ic-top-navigation --> ic-section-container
  ic-top-navigation --> ic-typography
  ic-top-navigation --> ic-button
  ic-top-navigation --> ic-horizontal-scroll
  ic-top-navigation --> ic-navigation-menu
  ic-button --> ic-loading-indicator
  ic-button --> ic-tooltip
  ic-loading-indicator --> ic-typography
  ic-tooltip --> ic-typography
  ic-horizontal-scroll --> ic-button
  ic-navigation-menu --> ic-button
  ic-navigation-menu --> ic-typography
  style ic-top-navigation fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


