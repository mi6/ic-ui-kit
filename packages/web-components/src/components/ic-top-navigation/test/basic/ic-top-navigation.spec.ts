import { newSpecPage } from "@stencil/core/testing";
import { DEVICE_SIZES } from "../../../../utils/helpers";
import { Button } from "../../../ic-button/ic-button";
import { NavigationItem } from "../../../ic-navigation-item/ic-navigation-item";
import { NavigationMenu } from "../../../ic-navigation-menu/ic-navigation-menu";
import { SearchBar } from "../../../ic-search-bar/ic-search-bar";
import { TopNavigation } from "../../ic-top-navigation";
import { TextField } from "../../../ic-text-field/ic-text-field";
import { waitForTimeout } from "../../../../testspec.setup";

const waitForSearchFocus = async () => {
  //delay to account for 100ms delay in search bar focus
  await waitForTimeout(200);
};

describe("ic-top-navigation", () => {
  it("renders with app title", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation app-title="ApplicationName"></ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-app-title");
  });

  it("renders with app icon & title", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation app-title="ApplicationName">
        <svg
          slot="app-icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0V0z" fill="none" />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
          />
        </svg>
      </ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-app-icon-and-title");
  });

  it("renders with custom title link url", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation app-title="ApplicationName" href="https://www.google.co.uk/">
      </ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-custom-title-link-url");
  });

  it("renders with Title, Version & Status", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation
        app-title="ApplicationName"
        status="alpha"
        version="v0.0.7"
      ></ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-title-version-status");
  });

  it("renders with search", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation
        app-title="ApplicationName"
        status="alpha"
        version="v0.0.7"
      >
      <ic-search-bar slot="search" label="Search" placeholder="Search">
      </ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-search");
  });

  it("renders with buttons", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation
        app-title="ApplicationName"
        status="alpha"
        version="v0.0.7"
      >
        <ic-navigation-button
          label="button1"
          slot="buttons"
          onclick="alert('test')"
        >
          <svg
            slot="icon"
            xmlns="http://www.w3.org/2000/svg"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            fill="#000000"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
            />
          </svg>
        </ic-navigation-button>
      </ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-buttons");
  });

  it("renders with navigation", async () => {
    const page = await newSpecPage({
      components: [TopNavigation, NavigationItem],
      html: `<ic-top-navigation
        app-title="ApplicationName"
        status="alpha"
        version="v0.0.7"
      >
        <ic-navigation-item label="Navigation 1" slot=navigation href="/"></ic-navigation-item>
      </ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-navigation");
  });

  it("tests app title updates", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation app-title="ApplicationName"></ic-top-navigation>`,
    });
    const appTitle = await page.root.shadowRoot.querySelector("h1");

    expect(appTitle.textContent).toEqual("ApplicationName");
    page.root.appTitle = "New Application Name";
    await page.waitForChanges();

    expect(appTitle.textContent).toEqual("New Application Name");
  });

  it("tests form in search slot", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation app-title="ApplicationName">
      <form role="search" slot="search">
        <ic-search-bar slot="search" label="Search" placeholder="Search">
        </ic-search-bar>
      </form>
      </ic-top-navigation>`,
    });

    expect(page.root).toMatchSnapshot("form-in-search-slot");
  });

  it("renders with slotted app-title content", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation
        status="alpha"
        version="v0.0.7"
      >
      <a slot="app-title" href="/">App Title</a>
      </ic-top-navigation>`,
    });
    expect(page.root).toMatchSnapshot("renders-with-slotted-app-title-content");
  });

  it("tests behaviour on small device", async () => {
    const page = await newSpecPage({
      components: [TopNavigation, Button, NavigationMenu, SearchBar, TextField],
      html: `<ic-top-navigation
      app-title="ApplicationName"
      status="alpha"
      version="v0.0.7"
    >
    <ic-search-bar label="Search" placeholder="Search" slot="search"></ic-search-bar>
    </ic-top-navigation>`,
    });

    await page.rootInstance.resizeObserverCallback(DEVICE_SIZES.S);
    await page.waitForChanges();

    let searchBtn = page.root.shadowRoot.querySelector(
      "#search-toggle-button"
    ) as HTMLElement;
    expect(searchBtn).not.toBeNull();

    searchBtn.click();
    await page.waitForChanges();
    await waitForSearchFocus();

    let menuButton = page.root.shadowRoot.querySelector(
      "#menu-button"
    ) as HTMLElement;
    expect(menuButton).not.toBeNull();
    expect(page.rootInstance.searchButtonClick).toBe(false);
    page.rootInstance.searchButtonMouseDownHandler();
    expect(page.rootInstance.searchButtonClick).toBe(true);

    searchBtn.click();
    await page.waitForChanges();
    await waitForSearchFocus();
    expect(page.rootInstance.searchButtonClick).toBe(false);

    menuButton.click();
    await page.waitForChanges();

    let menu = page.root.shadowRoot.querySelector("ic-navigation-menu");
    expect(menu).not.toBeNull();

    await page.rootInstance.resizeObserverCallback(DEVICE_SIZES.XL);
    await page.waitForChanges();

    menu = page.root.shadowRoot.querySelector("ic-navigation-menu");
    expect(menu).toBeNull();

    searchBtn = page.root.shadowRoot.querySelector(
      "#search-toggle-button"
    ) as HTMLElement;
    expect(searchBtn).toBeNull();
    menuButton = page.root.shadowRoot.querySelector(
      "#menu-button"
    ) as HTMLElement;
    expect(menuButton).toBeNull();
  });

  it("tests hiding of mobile search bar on resize", async () => {
    const page = await newSpecPage({
      components: [TopNavigation, Button, NavigationMenu, SearchBar, TextField],
      html: `<ic-top-navigation
      app-title="ApplicationName"
      status="alpha"
      version="v0.0.7"
    >
    <ic-search-bar label="Search" placeholder="Search" slot="search"></ic-search-bar>
    </ic-top-navigation>`,
    });

    await page.rootInstance.resizeObserverCallback(DEVICE_SIZES.S);
    await page.waitForChanges();
    const searchBtn = page.root.shadowRoot.querySelector(
      "#search-toggle-button"
    ) as HTMLElement;
    expect(searchBtn).not.toBeNull();

    searchBtn.click();
    await page.waitForChanges();
    await waitForSearchFocus();
    expect(page.rootInstance.mobileSearchBarVisible).toBe(true);

    await page.rootInstance.resizeObserverCallback(DEVICE_SIZES.XL);
    await page.waitForChanges();
    await waitForSearchFocus();
    expect(page.rootInstance.mobileSearchBarVisible).toBe(false);
  });

  it("should test nav bar menu visibility", async () => {
    const page = await newSpecPage({
      components: [TopNavigation, Button, NavigationMenu, SearchBar, TextField],
      html: `<ic-top-navigation
      app-title="ApplicationName"
      status="alpha"
      version="v0.0.7"
    >
    <ic-search-bar label="Search" placeholder="Search" slot="search"></ic-search-bar>
    </ic-top-navigation>`,
    });

    await page.rootInstance.resizeObserverCallback(DEVICE_SIZES.S);
    await page.waitForChanges();

    const menuButton = page.root.shadowRoot.querySelector(
      "#menu-button"
    ) as HTMLElement;
    expect(menuButton).not.toBeNull();
    expect(page.rootInstance.navMenuVisible).toBe(false);
    menuButton.click();
    await page.waitForChanges();
    expect(page.rootInstance.navMenuVisible).toBe(true);
    await page.rootInstance.navBarMenuCloseHandler();
    await page.waitForChanges();
    expect(page.rootInstance.navMenuVisible).toBe(false);
  });

  it("tests short title on mobile devices", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation
      app-title="Application Name"
      short-app-title="App Name"
    >
    </ic-top-navigation>`,
    });

    await page.rootInstance.resizeObserverCallback(DEVICE_SIZES.S);
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot("renders-with-short-title");
  });

  it("should test search value change", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation
      app-title="ApplicationName"
      status="alpha"
      version="v0.0.7"
    >
    </ic-top-navigation>`,
    });

    //test search value change
    await page.rootInstance.searchValueChangeHandler({
      detail: { value: "hi!" },
    });
    expect(page.rootInstance.searchValue).toBe("hi!");

    //test search input blur
    page.rootInstance.mobileSearchBarVisible = true;
    page.rootInstance.searchButtonClick = false;
    await page.rootInstance.searchInputBlurHandler({
      detail: { value: "hi again!" },
    });
    expect(page.rootInstance.searchValue).toBe("hi again!");
  });

  it("should test theme change", async () => {
    const page = await newSpecPage({
      components: [TopNavigation],
      html: `<ic-top-navigation
      app-title="ApplicationName"
      status="alpha"
      version="v0.0.7"
    >
    </ic-top-navigation>`,
    });

    await page.rootInstance.themeChangeHandler({ detail: { mode: "dark" } });
    await page.waitForChanges();
    expect(page.rootInstance.foregroundColor).toBe("dark");

    await page.rootInstance.themeChangeHandler({ detail: { mode: "light" } });
    await page.waitForChanges();
    expect(page.rootInstance.foregroundColor).toBe("light");

    //call runResizeObserver
    await page.rootInstance.runResizeObserver();

    //test disconnected callback
    page.setContent("");
  });
});
