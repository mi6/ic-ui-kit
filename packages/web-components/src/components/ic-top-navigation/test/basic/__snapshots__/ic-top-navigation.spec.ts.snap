// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-top-navigation renders with Title, Version & Status: renders-with-title-version-status 1`] = `
<ic-top-navigation app-title="ApplicationName" status="alpha" version="v0.0.7">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
              <div class="app-status">
                <ic-typography aria-label="app tag" class="app-status-text" variant="label-uppercase">
                  alpha
                </ic-typography>
              </div>
              <div class="app-version">
                <ic-typography aria-label="app version" class="app-version-text" variant="label">
                  v0.0.7
                </ic-typography>
              </div>
            </div>
            <div class="search-menu-container">
              <div class="search-actions-container">
                <slot name="search"></slot>
              </div>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
</ic-top-navigation>
`;

exports[`ic-top-navigation renders with app icon & title: renders-with-app-icon-and-title 1`] = `
<ic-top-navigation app-title="ApplicationName">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <div aria-hidden="true" class="app-icon-container">
                  <slot name="app-icon"></slot>
                </div>
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
  <svg fill="#000000" height="24px" slot="app-icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0h24v24H0V0z" fill="none"></path>
    <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"></path>
  </svg>
</ic-top-navigation>
`;

exports[`ic-top-navigation renders with app title: renders-with-app-title 1`] = `
<ic-top-navigation app-title="ApplicationName">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
</ic-top-navigation>
`;

exports[`ic-top-navigation renders with buttons: renders-with-buttons 1`] = `
<ic-top-navigation app-title="ApplicationName" status="alpha" version="v0.0.7">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
              <div class="app-status">
                <ic-typography aria-label="app tag" class="app-status-text" variant="label-uppercase">
                  alpha
                </ic-typography>
              </div>
              <div class="app-version">
                <ic-typography aria-label="app version" class="app-version-text" variant="label">
                  v0.0.7
                </ic-typography>
              </div>
            </div>
            <div class="search-menu-container">
              <div class="search-actions-container">
                <slot name="search"></slot>
                <div class="icon-buttons-container">
                  <slot name="buttons"></slot>
                </div>
              </div>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
  <ic-navigation-button label="button1" onclick="alert('test')" slot="buttons">
    <svg fill="#000000" height="24px" slot="icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M0 0h24v24H0V0z" fill="none"></path>
      <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"></path>
    </svg>
  </ic-navigation-button>
</ic-top-navigation>
`;

exports[`ic-top-navigation renders with custom title link url: renders-with-custom-title-link-url 1`] = `
<ic-top-navigation app-title="ApplicationName" href="https://www.google.co.uk/">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="https://www.google.co.uk/">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
</ic-top-navigation>
`;

exports[`ic-top-navigation renders with navigation: renders-with-navigation 1`] = `
<ic-top-navigation app-title="ApplicationName" status="alpha" version="v0.0.7">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
              <div class="app-status">
                <ic-typography aria-label="app tag" class="app-status-text" variant="label-uppercase">
                  alpha
                </ic-typography>
              </div>
              <div class="app-version">
                <ic-typography aria-label="app version" class="app-version-text" variant="label">
                  v0.0.7
                </ic-typography>
              </div>
            </div>
            <div class="search-menu-container">
              <div class="search-actions-container">
                <slot name="search"></slot>
              </div>
            </div>
          </div>
          <div class="navigation-tabs">
            <span aria-hidden="true" class="navigation-landmark-text" id="navigation-landmark-text">
              Main navigation
            </span>
            <nav aria-labelledby="navigation-landmark-text" class="nav-panel-container">
              <ic-horizontal-scroll appearance="light">
                <ul class="navigation-item-list" tabindex="-1">
                  <slot name="navigation"></slot>
                </ul>
              </ic-horizontal-scroll>
            </nav>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
  <ic-navigation-item class="light navigation-item navigation-item-top-nav" href="/" label="Navigation 1" role="listitem" slot="navigation">
    <mock:shadow-root>
      <ic-tooltip class="tooltip-navigation-item" label="Navigation 1" placement="right" target="navigation-item">
        <a class="link" href="/" part="link">
          <ic-typography variant="label">
            Navigation 1
          </ic-typography>
        </a>
      </ic-tooltip>
    </mock:shadow-root>
  </ic-navigation-item>
</ic-top-navigation>
`;

exports[`ic-top-navigation renders with search: renders-with-search 1`] = `
<ic-top-navigation app-title="ApplicationName" status="alpha" version="v0.0.7">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
              <div class="app-status">
                <ic-typography aria-label="app tag" class="app-status-text" variant="label-uppercase">
                  alpha
                </ic-typography>
              </div>
              <div class="app-version">
                <ic-typography aria-label="app version" class="app-version-text" variant="label">
                  v0.0.7
                </ic-typography>
              </div>
            </div>
            <div class="search-menu-container">
              <div class="search-actions-container">
                <slot name="search"></slot>
              </div>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
  <ic-search-bar label="Search" placeholder="Search" slot="search"></ic-search-bar>
</ic-top-navigation>
`;

exports[`ic-top-navigation renders with slotted app-title content: renders-with-slotted-app-title-content 1`] = `
<ic-top-navigation status="alpha" version="v0.0.7">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <div class="title-link">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    <slot name="app-title"></slot>
                  </h1>
                </ic-typography>
              </div>
              <div class="app-status">
                <ic-typography aria-label="app tag" class="app-status-text" variant="label-uppercase">
                  alpha
                </ic-typography>
              </div>
              <div class="app-version">
                <ic-typography aria-label="app version" class="app-version-text" variant="label">
                  v0.0.7
                </ic-typography>
              </div>
            </div>
            <div class="search-menu-container">
              <div class="search-actions-container">
                <slot name="search"></slot>
              </div>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
  <a href="/" slot="app-title">
    App Title
  </a>
</ic-top-navigation>
`;

exports[`ic-top-navigation tests form in search slot: form-in-search-slot 1`] = `
<ic-top-navigation app-title="ApplicationName">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <ic-typography variant="h3">
                  <h1 class="title-wrap">
                    ApplicationName
                  </h1>
                </ic-typography>
              </a>
            </div>
            <div class="search-menu-container">
              <div class="search-actions-container">
                <slot name="search"></slot>
              </div>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
  <form role="search" slot="search">
    <ic-search-bar label="Search" placeholder="Search" slot="search"></ic-search-bar>
  </form>
</ic-top-navigation>
`;

exports[`ic-top-navigation tests short title on mobile devices: renders-with-short-title 1`] = `
<ic-top-navigation app-title="Application Name" short-app-title="App Name">
  <mock:shadow-root>
    <div class="top-navigation">
      <ic-section-container aligned="full-width" full-height="">
        <header role="banner">
          <div class="top-panel-container">
            <div class="app-details-container">
              <a class="title-link" href="/">
                <ic-typography aria-label="Application Name (App Name)" variant="subtitle-small">
                  <h1>
                    App Name
                  </h1>
                </ic-typography>
              </a>
            </div>
          </div>
        </header>
      </ic-section-container>
    </div>
  </mock:shadow-root>
</ic-top-navigation>
`;
